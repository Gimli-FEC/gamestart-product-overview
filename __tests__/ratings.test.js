import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import $ from 'jquery';

import Ratings from '../client/components/ratings.jsx';


afterEach(cleanup);

jest.mock('$');

test('renders the component properly', () => {
  const { asFragment } = render(<Ratings />);
  expect(asFragment()).toMatchSnapshot();
});

test('loads tooltip when hovering over rating stars', () => {
  const result = render(<Ratings />);
  fireEvent.focus(result.getByTestId('tt'));
  expect(result.getByTestId('tt')).toBeInTheDocument();
});
