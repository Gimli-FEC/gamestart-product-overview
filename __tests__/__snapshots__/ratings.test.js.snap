// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="ratings__Row-sc-8kadw3-3 iEETze"
  >
    <div
      class="ratings__Col-sc-8kadw3-4 cwiHmS"
      size="1"
    >
      <div
        class="ratings__Container-sc-8kadw3-0 cbQoLz"
      >
        <span
          class="ratingsTooltip__Popup-sc-1d6glbo-0 irRflo"
        >
          <span
            class="ratingsTooltip__Trigger-sc-1d6glbo-1 bevQCd"
          >
            <div
              class="ratings__TopStars-sc-8kadw3-1 dEntyF"
            >
              ⭑⭑⭑⭑⭑
            </div>
            <div
              class="ratings__BottomStars-sc-8kadw3-2 iBSdYZ"
            >
              ⭑⭑⭑⭑⭑
            </div>
          </span>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
